### ISSUE TITLE
Example application

### ISSUE THREAD

**Issue by royrico** [jameswlane/jest-express#80]
Is there an example/demo application which we can look at.  The documentation provides the example code test cases, but not example code for what the application code looks like.

Thanks,
Roy

**Comment by jameswlane** [comment]
This is something that’s been on my list of TODOs. I used it a lot in testing for my project [Status Board](https://github.com/jameswlane/status-board) so you can dig through the test and production code. 

If you have any other questions, you can open a issue and message me on [Twitter](https://twitter.com/jameswlane)

If you would like to put together examples for the project pull request are always welcomed.

**Comment by lonix1** [comment]
I agree, this library is top quality, but without documentation it's hard to make sense of it.

**Comment by robertwbradford** [comment]
@jameswlane Great library. Thank you. I'm also wondering about this. Specifically on your examples, you have an `import { endpoint } from '../src/endpoint.js';`. This is then called in your tests with the instance of the Router, Response, or Request. What exactly is the `endpoint` function doing? Or what should it be doing?

**Comment by jameswlane** [comment]
@lonix1 I did my best to document everything in the [README.md](https://github.com/jameswlane/jest-express/), I am looking for any help with improving the documentation, get a website up with APIs/Docs/Examples, or get examples of testing certain things in express. So PR's are always welcome.

@robertwbradford I am normally thinking of a route or module that handles a response.
Here are some examples from one of my projects:
* [source](https://github.com/jameswlane/status-board/blob/master/src/webapp/routes.ts) => [test](https://github.com/jameswlane/status-board/blob/master/test/unit/webapp/routes.test.ts)
* [source](https://github.com/jameswlane/status-board/blob/master/src/webapp/routes/widget/render-js-widget.ts) => [test](https://github.com/jameswlane/status-board/blob/master/test/unit/webapp/routes/widget/render-js-widget.test.ts)
* [source](https://github.com/jameswlane/status-board/blob/master/src/webapp/server.ts) => [test](https://github.com/jameswlane/status-board/blob/master/test/unit/webapp/server.test.ts)

This project is what started `jest-express` but the test could be updated to better use everything that has been added since it was first written but they are good rough examples.

Let me know if there is anything I can help with, and if anyone would like to help with documentation, examples, or [website](https://github.com/jameswlane/jest.express). PR's are always welcome.

**Comment by HugoDF** [comment]
I've got an example repo for express request/response mocking at https://github.com/HugoDF/mock-express-request-response (and matching blog post but that's enough self-promotion)

I guess the patterns shown there can also leverage jest-express (although might need to be adapted)
